tests/suite/test_virtual_server_grpc.py

```diff
@@ -0,0 +1,178 @@
+            stub = GreeterStub(channel)
+            response = ""
+            try:
+                response = stub.SayHello(HelloRequest(name=virtual_server_setup.public_endpoint.public_ip))
+                print(response)
+            except grpc.RpcError as e:
+                print(e.details())
+                pytest.fail("RPC error was not expected during call, exiting...")
+
+    def patch_valid_vs(self, kube_apis, virtual_server_setup) -> None:
+        """
+        Function to revert vs deployment to valid state
+        """
+        patch_src = f"{TEST_DATA}/virtual-server-grpc/standard/virtual-server.yaml"
+        patch_virtual_server_from_yaml(
+            kube_apis.custom_objects,
+            virtual_server_setup.vs_name,
+            patch_src,
+            virtual_server_setup.namespace,
+        )
+
+    @pytest.mark.parametrize("backend_setup", [{"app_type": "grpc-vs"}], indirect=True)
+    def test_config_after_setup(self, kube_apis, ingress_controller_prerequisites, crd_ingress_controller, 
+                                backend_setup, virtual_server_setup):
+        print("\nStep 1: assert config")
+        ic_pod_name = get_first_pod_name(kube_apis.v1, ingress_controller_prerequisites.namespace)
+        config = get_vs_nginx_template_conf(kube_apis.v1,
+                                            virtual_server_setup.namespace,
+                                            virtual_server_setup.vs_name,
+                                            ic_pod_name,
+                                            ingress_controller_prerequisites.namespace)
+
+        assert "proxy_connect_timeout 60s;" not in config
+        assert "proxy_read_timeout 60s;" not in config
+        assert "proxy_send_timeout 60s;" not in config
+
+        assert "proxy_set_header Upgrade $http_upgrade;" not in config
+        assert "proxy_http_version 1.1;" not in config
+
+        assert "proxy_next_upstream error timeout;" not in config
+        assert "proxy_next_upstream_timeout 0s;" not in config
+        assert "proxy_next_upstream_tries 0;" not in config
+
+        assert "grpc_connect_timeout 60s;" in config
+        assert "grpc_read_timeout 60s;" in config
+        assert "grpc_send_timeout 60s;" in config
+
+        assert "grpc_set_header X-Real-IP $remote_addr;" in config
+        assert "grpc_set_header X-Forwarded-For $proxy_add_x_forwarded_for;" in config
+        assert "grpc_set_header X-Forwarded-Host $host;" in config
+        assert "grpc_set_header X-Forwarded-Port $server_port;" in config
+        assert "grpc_set_header X-Forwarded-Proto $scheme;" in config
+
+        assert 'grpc_set_header Host "$host";' in config
+
+        assert "grpc_next_upstream error timeout;" in config
+        assert "grpc_next_upstream_timeout 0s;" in config
+        assert "grpc_next_upstream_tries 0;" in config
+
+        print("\nStep 2: Check connection to app")
+        self.connect_grpc_backend(virtual_server_setup)
```

Comment:
is it possible to assert something based on the data in response?

Reply:
Done